### ISSUE TITLE
qutebroswer prevents the display from going to sleep

### ISSUE THREAD

**Issue by lufte** [qutebrowser/qutebrowser#3833]
Hi, I'm running the debian packaged version of qutebrowser and I have found out that sometimes it prevents the display from sleeping/switching off.

DPMS is handled by xfce4-power-manager. From what I've been able to test, you need to play a video on qutebrowser (say, from youtube) for it to inhibit display blanking, which makes sense, but after closing the tab it doesn't restore the previous state.

```
qutebrowser v1.2.1
Git commit: 
Backend: QtWebEngine (Chromium 56.0.2924.122)

CPython: 3.6.5rc1
Qt: 5.9.2
PyQt: 5.9.2

sip: 4.19.8
colorama: no
pypeg2: 2.15
jinja2: 2.10
pygments: 2.2.0
yaml: 3.12
cssutils: no
attr: 17.4.0
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebKitWidgets: no
pdf.js: no
sqlite: 3.22.0
QtNetwork SSL: OpenSSL 1.0.2o  27 Mar 2018

Style: QFusionStyle
Platform: Linux-4.15.0-2-amd64-x86_64-with-debian-buster-sid, 64bit
Linux distribution: Debian GNU/Linux buster/sid (debian)
Frozen: False
Imported from /usr/lib/python3/dist-packages/qutebrowser
Qt library executable path: /usr/lib/x86_64-linux-gnu/qt5/libexec, data path: /usr/share/qt5

Paths:
cache: /home/xavi/.cache/qutebrowser
config: /home/xavi/.config/qutebrowser
data: /home/xavi/.local/share/qutebrowser
runtime: /run/user/1000/qutebrowser
system data: /usr/share/qutebrowser

Uptime: 0:02:19
```

**Comment by The-Compiler** [comment]
Can you please check whether you see the same behavior with [KDE Falkon](https://github.com/KDE/falkon)? Unfortunately it's not packaged in Debian (only it's predecessor, QupZilla, and only in a too old version to test this), but it should be straightforward to build (see its readme).

**Comment by lufte** [comment]
I'll test and report back.

**Comment by The-Compiler** [comment]
@lufte Any update on this?

**Comment by lufte** [comment]
Hi Florian, it turns out the reproduction steps were not as straightforward
as I initially posted but now I'm gonna be away from home for a couple of
weeks so I can't continue testing. I'll try to update as soon as I get back.

**Comment by lufte** [comment]
I found these steps which seem to work every time. I was initially able to reproduce it on Falkon but these steps don't seem to work on it, only qutebrowser.

1. Check that DPMS is working and your screen turns off.
2. Open a new instance of qutebrowser and visit a youtube video. Make sure it plays for a few seconds.
3. Now in the same tab open hackernews: https://news.ycombinator.com/news.
4. Wait until the display is supposed to turn off.

This is my updated version info:
```
qutebrowser v1.2.1
Git commit: 
Backend: QtWebEngine (Chromium 61.0.3163.140)

CPython: 3.6.5rc1
Qt: 5.10.1
PyQt: 5.10.1

sip: 4.19.8
colorama: no
pypeg2: 2.15
jinja2: 2.10
pygments: 2.2.0
yaml: 3.12
cssutils: no
attr: 17.4.0
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebKitWidgets: no
pdf.js: no
sqlite: 3.23.1
QtNetwork SSL: OpenSSL 1.1.0h  27 Mar 2018

Style: QFusionStyle
Platform: Linux-4.16.0-1-amd64-x86_64-with-debian-buster-sid, 64bit
Linux distribution: Debian GNU/Linux buster/sid (debian)
Frozen: False
Imported from /usr/lib/python3/dist-packages/qutebrowser
Qt library executable path: /usr/lib/x86_64-linux-gnu/qt5/libexec, data path: /usr/share/qt5

Paths:
cache: /home/xavi/.cache/qutebrowser
config: /home/xavi/.config/qutebrowser
data: /home/xavi/.local/share/qutebrowser
runtime: /run/user/1000/qutebrowser
system data: /usr/share/qutebrowser

Uptime: 0:13:32
```

**Comment by cvincent** [comment]
I've also experienced this. It can be frustrating because I run multiple Qutebrowser instances to sandbox tasks, and figuring out which instance is keeping my display awake is a matter of trial and error by closing individual instances and seeing if the display turns off or not. I'd actually be happy to just be able to more easily to figure out which instance/tab/website is keeping the display awake.

**Comment by cvincent** [comment]
I've come up with a rather hacky workaround in the meantime. I've installed `xscreensaver`, picked an arbitrary screensaver, went to its advanced settings, and added `xset dpms force off;` to the command (the semicolon is important to swallow parameters added by `xscreensaver`, as this isn't a proper screensaver command). Then I've set "Blank after" to 5 minutes. Optionally, I've also set "Lock after" to 0 minutes.

The result is, after 5 minutes of inactivity, `xscreensaver` will turn off your display(s). It will also immediately lock the screen, so when you hit a key or move your mouse, you'll need to enter your password to get back in.

According to ArchWiki, some apps will disable `xscreensaver`. For example `mpv` (which I usually use to watch videos) will prevent `xscreensaver` from triggering while a video is playing. This is desirable behavior. Yet, importantly, no Qutebrowser instance disables `xscreensaver` in my testing, so I don't have to worry about some tab or Qutebrowser instance quietly sabotaging my monitors.

I haven't tested whether, e.g., Steam games disable it, which is something you'd want. YMMV. For the time being, this prevents my displays from staying lit unexpectedly for long stretches of time and is a decent stopgap until a better solution is found. More generally, this isn't the first time I've had issues with DPMS, and from my past searching, it unfortunately seems to be an area where Linux really needs improvement.

**Comment by cvincent** [comment]
Okay so turns out my workaround isn't sufficient, as I return to my workstation to find the screens blanked but backlights still on. Appears related to this shockingly long-standing bug: https://bugs.launchpad.net/ubuntu/+source/gnome-session/+bug/1600622

**Comment by The-Compiler** [comment]
@lufte @cvincent Can you still reproduce this with more recent versions of Qt?

**Comment by lufte** [comment]
I can still reproduce it with Qt 5.11.3.

**Comment by edio** [comment]
Found this bug, while searching for related issue.
For me it is the opposite. While watching youtube, screen goes off according to powersaving settings. I'd expect it remain on, if video is playing.

**Comment by The-Compiler** [comment]
@edio Can't say much without seeing any version information. If you're using QtWebEngine, please also try how things look in [KDE Falkon](https://github.com/KDE/falkon).

**Comment by edio** [comment]
@The-Compiler I didn't post any details because I wasn't sure, whether I'll be redirected to another issue.

However, I tested just now, and the issue is reproduced on qutebrowser, falkon and firefox. So I guess the issue is KDE's powerdevil, not browser. I should've checked before posting here, sorry.

**Comment by The-Compiler** [comment]
@edio No worries!

**Comment by The-Compiler** [comment]
@lufte @cvincent Sorry for the radio silence on this one - is this something you still see nowadays?

**Comment by cvincent** [comment]
These days I'm using a slightly different setup (i3 via Regolith as opposed to i3 on its own) and haven't had this issue, no special setup or workaround required. I don't know if that's chalked up to my DE or what.

**Comment by lufte** [comment]
Yes, I just tried and it still occurs. It works as expected if you close the tab, but if you navigate to a different website it will keep the screen awake.

**Comment by ncoder-1** [comment]
I can confirm the same behaviour on my side, running Arch and i3-gaps.  At first I thought it was Steam, closed it and waited for the timer (using [xidlehook](https://github.com/jD91mZM2/xidlehook)) to engage my screenlock ([betterlockscreen](https://github.com/pavanjadhaw/betterlockscreen)).  Never went off.  So I did the opposite, closed qutebrowser and left Steam open, the timer worked and my screen locked.  

So I checked further, and it is website dependent.  For example, if I open reddit on the frontpage, posts that have "live previews" such as a livestream or a video, will somehow prevent the lock from ever launching, even with no audio playing.  A static reddit page will be fine and the lock will go off.

The proper behaviour, I believe, would be that the inhibitor should only be active when 
- sound is playing && that tab is focused
- or, video is playing fullscreen

Right now it looks like there's an inhibitor when
- video/animation is playing whether the tab is focused or not

My idlehook launch command (from my i3 config file):
`exec --no-startup-id xidlehook --not-when-fullscreen --not-when-audio --timer 120 "betterlockscreen -l dimblur --off 5" ''`

**Comment by The-Compiler** [comment]
@ncoder-1 Note there's nothing in qutebrowser directly doing this - it's all coming from the underlying QtWebEngine and/or Chromium. If you have a reliable reproducer for this, could you please check whether you can reproduce with KDE Falkon and Chromium as well?

**Comment by ncoder-1** [comment]
I just did a few rounds of testing with Chromium.  I was expecting the same behavior with the same websites, unfortunately, I can't reproduce the issue with Chromium (the lock screen appears on time), but I can reproduce the issues perfectly with qutebrowser.

I 100% believe you that there's nothing in qutebrowser directly doing this, I'm starting to wonder is there's a filter somewhere in i3 or picom, maybe class id-based.  I'll give Falkon a shot.
